pgmpy/estimators/MLE.py

```diff
@@ -48,38 +50,80 @@ def get_parameters(self):
-                                 evidence_card=parent_card.astype('int'))
-                cpd.normalize()
-                parameters.append(cpd)
+            cpd = self._estimate_cpd(node)
+            parameters.append(cpd)
 
         return parameters
+
+    def _estimate_cpd(self, node):
+        """
+        Method to estimate the CPD for a given variable.
+
+        Parameters
+        ----------
+        node: int, string (any hashable python object)
+            The name of the variable for which the CPD is to be estimated.
+
+        Returns
+        -------
+        CPD: TabularCPD
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> from pgmpy.models import BayesianModel
+        >>> from pgmpy.estimators import MaximumLikelihoodEstimator
+        >>> data = pd.DataFrame(data={'A': [0, 0, 1], 'B': [0, 1, 0], 'C': [1, 1, 0]})
+        >>> model = BayesianModel([('A', 'C'), ('B', 'C')])
+        >>> cpd_A = MaximumLikelihoodEstimator(model, data)._estimate_cpd('A')
+        >>> print(str(cpd_A))
+        ╒══════╤══════════╕
+        │ A(0) │ 0.666667 │
+        ├──────┼──────────┤
+        │ A(1) │ 0.333333 │
+        ╘══════╧══════════╛
+        """
+
+        parents = sorted(self.model.get_parents(node))
+        node_cardinality = len(self.node_values[node])
+        parents_cardinalities = np.array([len(self.node_values[parent]) for parent in parents])
+
+        if not parents:
+            state_count_data = self.data.ix[:, node].value_counts()
+            state_counts = state_count_data.reindex(sorted(self.node_values[node])).fillna(0).values[:, np.newaxis]
+
+        else:
+            state_count_data = self.data.groupby([node] + parents).size()
+            state_counts = state_count_data.unstack(parents).reindex(sorted(self.node_values[node])).fillna(0)
+            if isinstance(state_counts.index, pd.MultiIndex):
+                state_counts = state_counts.sortlevel(axis=1)
+            else:
+                state_counts = state_counts.sort_index(axis=1)
+
+            # some columns might be missing if for some states of the parents no data was observed:
+            if not len(state_counts.columns) == np.prod(parents_cardinalities):
+                possible_parents_states = [sorted(self.node_values[parent]) for parent in parents]
+                # reindex to add missing columns and fill in uniform (conditional) probabilities:
+                full_index = pd.MultiIndex.from_product(possible_parents_states, names=parents)
+                state_counts = state_counts.reindex(columns=full_index).fillna(1.0 / node_cardinality)
+
+        state_names = {var: sorted(states) for var, states in self.node_values.items()}
+        cpd = TabularCPD(node, node_cardinality, np.array(state_counts),
+                         evidence=parents,
+                         evidence_card=parents_cardinalities,
+                         state_names=state_names)
+        cpd.normalize()
+        return cpd
```

Comment:
Hm, I'm positive that the files end with a newline. When i add another one my PEP8 linter complains about having an extra blank line in the end.